backstory/src/test/java/com/google/sps/story/StoryManagerTest.java

```diff
@@ -144,4 +150,23 @@ public void correctRequestConfiguration() throws RuntimeException {
     Assert.assertEquals(expectedRequestString, actualRequestString);
     Assert.assertEquals(expectedOutput, actualOutput);
   }
+
+  @Test
+  /**
+   * Ensures Provider does not cycle past bounds.
+   */
+  public void cyclePastBounds() {
+    try {
+      String firstURL = URLProvider.getCurrentURL();
+      URLProvider.cycleURL();
```

Comment:
I think you could use a do-while construct to avoid needing this statement twice:
https://docs.oracle.com/javase/tutorial/java/nutsandbolts/while.html

Reply:
Done